.github/workflows/publish-to-pypi.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish Modin wheel to PyPI
+
+on: push
+
+jobs:
+  build-n-publish:
+    name: Build and publish Modin wheel to PyPI
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: "3.9.x"
+
+    - name: Install/update tools
+      run: python3 -m pip install --upgrade build wheel
+    - name: Build a pure Python wheel
+      run: python3 setup.py sdist bdist_wheel
+
+    - uses: actions/upload-artifact@v4
+      with:
+        name: modin-wheel-and-source-tarball 
+        path: ./dist/
+
+    - name: Publish Modin wheel to PyPI
+      if: startsWith(github.ref, 'refs/tags')
+      uses: pypa/gh-action-pypi-publish@release/v1
+      with:
+        password: ${{ secrets.PUBLISH_TO_PYPI_TOKEN }}
```

Comment:
For this flow to work, we need to add a token.

Reply:
We can just add [a publisher to PyPI](https://docs.pypi.org/trusted-publishers/adding-a-publisher/) so there is no need for a token.

Reply:
Added trusted publisher since it is best practice regardless of if we use token.